src/sentry/dynamic_sampling/models/transaction_adjustment_model.py

```diff
@@ -0,0 +1,190 @@
+from collections import namedtuple
+from copy import copy
+from typing import List, Mapping, MutableMapping, Tuple
+
+AdjustedSampleRate = namedtuple("AdjustedSampleRate", "explicit_rates, global_rate")
+
+
+def adjust_sample_rate(
+    transactions: List[Tuple[str, int]], rate: float, max_explicit_transactions: int
+) -> Tuple[Mapping[str, float], float]:
+    """
+    Calculates transaction sample size in order to maximize the number of small transactions
+
+    :param transactions: the transaction types as an array of (name, count) tuples
+    :param rate: the overall desired rate
+    :param max_explicit_transactions: the maximum number of transactions that can have individual
+        rates set, the rest will have a common rate
+    :return: a tuple with the first element a mapping transaction-name->sampling-rate and the
+    second element the transaction rate for all other transactions (that are not in the dict)
+    """
+    # sort by transaction count
+    transactions = sorted(transactions, key=lambda x: x[1])
```

Comment:
```suggestion
    import operator
    transactions = sorted(transactions, key=operator.itemgetter(1))
```

Reply:
just a suggestion they are both equivalent I don't have a strong opinion on it.

Reply:
cool, didn't know about this